server/container_create.go

```diff
@@ -495,26 +495,49 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 		capabilities := linux.GetSecurityContext().GetCapabilities()
 		toCAPPrefixed := func(cap string) string {
 			if !strings.HasPrefix(strings.ToLower(cap), "cap_") {
-				return "CAP_" + cap
+				return "CAP_" + strings.ToUpper(cap)
 			}
 			return cap
 		}
+
+		// Add/drop all capabilities if "all" is specified, so that
+		// following individual add/drop could still work. E.g.
+		// AddCapabilities: []string{"ALL"}, DropCapabilities: []string{"CHOWN"}
+		// will be all capabilities without `CAP_CHOWN`.
+		// see https://github.com/kubernetes/kubernetes/issues/51980
+		if inStringSlice(capabilities.GetAddCapabilities(), "ALL") {
```

Comment:
Does this work, isn't Capabilities a list at this point?

Reply:
It is yeah, slice is a list, I don't get the question though

Reply:
This code works, tested it with kube directly

Reply:
capabilities.GetAddCapabilities()

If this is a slice how does it have a method associated with it.  I thought capabilities was already of list of everything specified by the caller.

Reply:
it's gRPC methods. Capabilities is an object with GetAddCapapibilities method which returns a slice..

Reply:
Ok below you are using

  			addCaps := capabilities.AddCapabilities 

And here you are using 
  			addCaps := capabilities.GetAddCapabilities()

Which is confusing me, but I guess they do the same thing.  We should probably be consistent.

Reply:
fixed to always use `capabilities.GetAddCapabilities()`